smt.py

```diff
@@ -237,12 +278,32 @@ def emit_expr(expr: source.ExprT[assume_prove.VarName]) -> SMTLIB:
             raise NotImplementedError(
                 "PAlignValid for non symbols isn't supported")
 
+        if expr.operator is source.Operator.MEM_ACC:
+            mem, symb_or_addr = expr.operands
+            if not isinstance(symb_or_addr, source.ExprSymbol):
+                raise NotImplementedError(
+                    "MemAcc for non symbols isn't supported yet")
+
+            if not isinstance(symb_or_addr.typ, source.TypeBitVec):
+                assert False, "Only TypBitVec is accepted"
+
+            as_fn_call = emit_expr(symb_or_addr)
+
+            if symb_or_addr.typ.size not in load_word_map.keys():
+                raise NotImplementedError(
+                    f"MemAcc for BitVec of size {symb_or_addr.typ.size} is not supported")
+            return SMTLIB(f"({load_word_map[symb_or_addr.typ.size]} {emit_expr(mem)} {as_fn_call})")
+
         return SMTLIB(f'({ops_to_smt[expr.operator]} {" ".join(emit_expr(op) for op in expr.operands)})')
     elif isinstance(expr, source.ExprVar):
         return SMTLIB(f'{identifier(expr.name)}')
-    elif isinstance(expr, source.ExprSymbol | source.ExprType):
+    elif isinstance(expr, source.ExprSymbol):
+        return emit_expr_symbol(expr)
```

Comment:
Shouldn't we declare-func the symbols ourselves (as in, in Python) rather than rely on them being in the prelude?